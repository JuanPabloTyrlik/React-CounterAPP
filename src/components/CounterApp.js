import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 0 }) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1);
    };

    const handleSubstract = () => {
        setCounter(counter - 1);
    };

    const handleReset = () => {
        setCounter(value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd} style={{width: '55px'}}>+1</button>
            <button onClick={handleSubstract} style={{width: '55px'}}>-1</button>
            <button onClick={handleReset}>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number,
};

export default CounterApp;
