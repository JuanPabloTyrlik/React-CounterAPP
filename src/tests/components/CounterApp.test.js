import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../../components/CounterApp';

describe('Pruebas de CounterApp', () => {
    let wrapper = shallow(<CounterApp />);
    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });
    test('should render <CounterApp /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should render <CounterApp /> correctly with counter 100', () => {
        expect(
            shallow(<CounterApp value={100} />)
                .find('h2')
                .text()
        ).toBe('100');
    });
    test('should increse counter by 1', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(parseInt(wrapper.find('h2').text())).toBe(1);
    });
    test('should increse counter by N', () => {
        const N = Math.floor(Math.random() * 10);
        for (let i = 0; i < N; i++) {
            wrapper.find('button').at(0).simulate('click');
        }
        expect(parseInt(wrapper.find('h2').text())).toBe(parseInt(N));
    });
    test('should decrease counter by 1', () => {
        wrapper.find('button').at(1).simulate('click');
        expect(parseInt(wrapper.find('h2').text())).toBe(-1);
    });
    test('should decrease counter by N', () => {
        const N = Math.floor(Math.random() * 10);
        for (let i = 0; i < N; i++) {
            wrapper.find('button').at(1).simulate('click');
        }
        expect(parseInt(wrapper.find('h2').text())).toBe(parseInt(N * -1));
    });
    test('should reset counter to 0', () => {
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        expect(parseInt(wrapper.find('h2').text())).toBe(0);
    });
    test('should reset counter to N', () => {
        const N = Math.floor(Math.random() * 10);
        const wrapper = shallow(<CounterApp value={N} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        expect(parseInt(wrapper.find('h2').text())).toBe(N);
    });
});
