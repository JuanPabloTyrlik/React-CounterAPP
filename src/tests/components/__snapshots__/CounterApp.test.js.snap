// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp should render <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
    style={
      Object {
        "width": "55px",
      }
    }
  >
    +1
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "width": "55px",
      }
    }
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
